### PR TITLE
bug-fix: save() all field cube (when no operation)

### PR DESCRIPTION
**Pull request by volodia99**
when doing ds[field].save(), updated the name of header file and npy file, without changing the process when doing ds[field].operation().save()

### CODE REVIEW

File: tests/test_load_data.py [hunk 1]
```diff
@@ -29,6 +30,21 @@ def test_roundtrip_simple(test_data_dir, tmp_path):
     assert len(list(dsnpy.keys())) == 1
 
 
+def test_roundtrip_no_operation_all_field(test_data_dir, tmp_path):
+    copytree(test_data_dir / "idefix_spherical_planet3d", tmp_path / "mydir")
```
Comment (neutrinoceros) [inline tests/test_load_data.py:34]: wouldn't it be more efficient to cd into the data dir and write *just* outputs to a tmp dir ?

File: tests/test_load_data.py [hunk 2]
```diff
@@ -29,6 +30,21 @@ def test_roundtrip_simple(test_data_dir, tmp_path):
...
+    os.chdir(tmp_path / "mydir")
+    ds = GasDataSet(500)
+    assert len(list(ds.keys())) == 7
+
+    gf = ds["RHO"]
+
+    gf.save()
+    dsnpy = GasDataSet.from_npy(500, operation="")
+    assert len(list(dsnpy.keys())) == 1
+    np.testing.assert_array_almost_equal(ds["RHO"].data, dsnpy["RHO"].data)
```
Comment (neutrinoceros) [inline tests/test_load_data.py:45]: Maybe there's something I'm missing and this doesn't work, but I would naively assume roundtrips preserves data exactly ?
```suggestion
    np.testing.assert_array_equal(ds["RHO"].data, dsnpy["RHO"].data)
```

File: tests/test_load_data.py [hunk 3]
```diff
@@ -29,6 +30,21 @@ def test_roundtrip_simple(test_data_dir, tmp_path):
     assert len(list(dsnpy.keys())) == 1
 
 
+def test_roundtrip_no_operation_all_field(test_data_dir, tmp_path):
+    copytree(test_data_dir / "idefix_spherical_planet3d", tmp_path / "mydir")
+
+    os.chdir(tmp_path / "mydir")
+    ds = GasDataSet(500)
+    assert len(list(ds.keys())) == 7
```
Comment (neutrinoceros) [inline tests/test_load_data.py:38]: Tangent, bu isn't there a more elegant way to do this ? I feel like there should be. Maybe something like `assert ds.nfields == 7` with
```python
@property
def nfields(self) -> int:
    return len(self.keys())
```

File: nonos/api/analysis.py [hunk 4]
```diff
@@ -600,13 +600,14 @@ def map(
         return Plotable(dict_plotable)
 
     def save(self, directory="", header_only=False) -> None:
+        operation = "_" if self.operation == "" else self.operation
```
Comment (neutrinoceros) [inline nonos/api/analysis.py:603]: This should be equivalent
```suggestion
        operation = self.operation or "_"
```

Comment (neutrinoceros) [reply nonos/api/analysis.py:603]: (Though I don't get this change)

File: tests/test_load_data.py (hunk 1 above)
Comment (volodia99) [reply tests/test_load_data.py:34]: maybe! In practice I used the structure of the `test_roundtrip_simple` function which is similar (I'm not familiar with copytree so I assumed it was from you)

File: nonos/api/analysis.py (hunk 4 above)
Comment (volodia99) [reply nonos/api/analysis.py:603]: oh wow indeed but for me it's a mystery why

**Pull request merged**